src/cluster.ts

```diff
@@ -399,34 +401,36 @@ Please use the format 'my-cluster' or '${instance.name}/clusters/my-cluster'.`);
    * region_tag:bigtable_cluster_set_meta
    */
   setMetadata(
-    metadata: CreateClusterOptions,
+    metadata: SetClusterMetadataOptions,
     gaxOptionsOrCallback?: CallOptions | SetClusterMetadataCallback,
     cb?: SetClusterMetadataCallback
   ): void | Promise<SetClusterMetadataResponse> {
     const callback =
       typeof gaxOptionsOrCallback === 'function' ? gaxOptionsOrCallback : cb!;
     const gaxOptions =
-      typeof gaxOptionsOrCallback === 'object' && gaxOptionsOrCallback
+      typeof gaxOptionsOrCallback === 'object'
         ? gaxOptionsOrCallback
         : ({} as CallOptions);
 
     const reqOpts: ICluster = {
       name: this.name,
+      serveNodes: metadata.nodes,
     };
 
-    if (metadata.location) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    if ((metadata as any).location) {
```

Comment:
If the API actually drops these parameters, I think we can skip all of the logic where we worry about formatting them.